libraries/hatch-server-user-management/src/UserManagementController.ts

```diff
@@ -0,0 +1,208 @@
+  public redirect(params: BasicRouteParams) {
+    params.res.redirect('/api');
+  }
+  
+  @route.post('/api/authenticate', AuthenticateRequest.apiMetadata)
+  public async authenticate(params: BasicRouteParams) {
+    this.logger.debug('Authenticating...');
+    try {
+      const {username, password} = params.req.body;
+      if (!username || !password || username.length === 0 || password.length === 0) {
+        const errMsg = 'Missing required field(s), username and password are required';
+        this.logger.debug(errMsg);
+        params.res.status(400).send({
+          error: errMsg,
+        });
+      } else {
+        const authTokens = await this.userService.authenticate(username, password);
+        params.res.cookie(AUTH_ACCESS_TOKEN_COOKIE_NAME, authTokens.accessToken);
+        this.logger.debug('User authenticated and cookie set');
+        params.res.status(200).send(authTokens);
+      }
+    } catch (err) {
+      this.logger.error('Error authenticating user: ', err);
+      params.res.status(500).send({
+        error: err,
+      });
+    }
+  }
+  
+  @route.post('/api/signUpUser', CreateUserRequest.apiMetadata)
+  public async signUpUser(params: BasicRouteParams) {
+    this.logger.debug('Creating user...');
+    try {
+      const {username, password, userAttributes} = params.req.body;
+      if (!username || !password || username.length === 0 || password.length === 0) {
+        const errMsg = 'Missing required field(s), username and password are required';
+        this.logger.debug(errMsg);
+        params.res.status(400).send({
+          error: errMsg,
+        });
+      } else {
+        await this.userService.signUpUser(username, password, userAttributes);
+        this.logger.debug('User created');
+        params.res.sendStatus(200);
+      }
+    } catch (err) {
+      this.logger.error('Error creating user: ', err);
+      params.res.status(500).send({
+        error: err,
+      });
+    }
+  }
+  
+  @route.post('/api/confirmUser', ConfirmUserRequest.apiMetadata)
+  public async confirmUser(params: BasicRouteParams) {
+    this.logger.debug('Confirming user...');
+    try {
+      const {username, confirmationCode} = params.req.body;
+      if (!username || !confirmationCode || username.length === 0 || confirmationCode.length === 0) {
+        const errMsg = 'Missing required field(s), username and confirmationCode are required';
+        this.logger.debug(errMsg);
+        params.res.status(400).send({
+          error: errMsg,
+        });
+      } else {
+        await this.userService.confirmUser(username,confirmationCode);
+        this.logger.debug('User confirmed');
+        params.res.sendStatus(200);
+      }
+    } catch (err) {
+      this.logger.error('Error confirming user: ', err);
+      params.res.status(500).send({
+        error: err,
+      });
+    }
+  }
+  
+  @route.post('/api/resetPassword', ResetPasswordRequest.apiMetadata)
+  public async resetPassword(params: BasicRouteParams) {
```

Comment:
Doesn't there need to be another endpoint for submitting the reset password code received via email?

Reply:
seems like there should be analogs for forgotPasswordSubmit, signOut, updateUserAttributes, resendSignUp

Reply:
Updated